src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftDeployer.java

```diff
@@ -69,47 +79,81 @@ public BuildStepMonitor getRequiredMonitorService() {
 		return null;
 	}
 
-    private static final Logger LOGGER = Logger.getLogger(OpenShiftDeployer.class.getName());
+	private static final Logger LOGGER = Logger.getLogger(OpenShiftDeployer.class.getName());
 
 
 	@Extension
-    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
-
-        public DescriptorImpl() {
-            super(OpenShiftDeployerExecution.class);
-        }
-
-        @Override
-        public String getFunctionName() {
-            return "openshiftDeploy";
-        }
-
-        @Override
-        public String getDisplayName() {
-            return DISPLAY_NAME;
-        }
-
-        @Override
-        public Step newInstance(Map<String, Object> arguments) throws Exception {
-            if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            Object depCfg = arguments.get("deploymentConfig");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	depCfg = arguments.get("depCfg");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            OpenShiftDeployer step = new OpenShiftDeployer(depCfg.toString());
-            
-            if (arguments.containsKey("waitTime")) {
-            	Object waitTime = arguments.get("waitTime");
-            	if (waitTime != null)
-            		step.setWaitTime(waitTime.toString());
-            }
-            
-            ParamVerify.updateDSLBaseStep(arguments, step);
-            return step;
-        }
-    }
+	public static class DescriptorImpl extends AbstractStepDescriptorImpl implements IOpenShiftTimedStepDescriptor {
+		private String wait;
+		private String waitUnit = "sec";
+
+		public DescriptorImpl() {
+			super(OpenShiftDeployerExecution.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "openshiftDeploy";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return DISPLAY_NAME;
+		}
+
+		@Override
+		public Step newInstance(Map<String, Object> arguments) throws Exception {
+			if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			Object depCfg = arguments.get("deploymentConfig");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				depCfg = arguments.get("depCfg");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			OpenShiftDeployer step = new OpenShiftDeployer(depCfg.toString());
+
+			if (arguments.containsKey("waitTime")) {
+				Object waitTime = arguments.get("waitTime");
+				if (waitTime != null) {
+					wait = waitTime.toString();
+					step.setWaitTime(wait);
+				}
+			}
+
+			if (arguments.containsKey("waitUnit")) {
```

Comment:
leverage the new default method on IOpenShiftTimedStepDescriptor called for earlier